src/lib/operators/table_scan.cpp

```diff
@@ -0,0 +1,290 @@
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "abstract_operator.hpp"
+#include "all_type_variant.hpp"
+#include "resolve_type.hpp"
+#include "storage/dictionary_segment.hpp"
+#include "storage/reference_segment.hpp"
+#include "storage/value_segment.hpp"
+#include "table_scan.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
+                     const ScanType scan_type, const AllTypeVariant search_value)
+    : _in{in}, _column_id{column_id}, _scan_type{scan_type}, _search_value{search_value} {}
```

Comment:
I would suggest not to store `_in` inside this subclass of `AbstractOperator`. This would do the job without an additional field `_in`: `TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in, const ColumnID column_id,
                     const ScanType scan_type, const AllTypeVariant search_value)
    : AbstractOperator(in), _column_id(column_id), _scan_type(scan_type), _search_value(search_value) {}`

Reply:
Good note, this is the proper way to deal with derived classes.